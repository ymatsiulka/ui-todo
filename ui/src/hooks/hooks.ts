import { useDispatch, useSelector, useStore, TypedUseSelectorHook } from 'react-redux';
import { AppDispatch, RootState, AppStore } from 'store/store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export const useAppStore: () => AppStore = useStore;
