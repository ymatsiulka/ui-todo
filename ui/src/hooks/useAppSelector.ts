import { useSelector, type TypedUseSelectorHook } from 'react-redux';
import { type RootState } from 'store/store';

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
