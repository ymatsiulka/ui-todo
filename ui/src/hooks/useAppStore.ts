import { useStore } from 'react-redux';
import { type AppStore } from 'store/store';

export const useAppStore: () => AppStore = useStore;
