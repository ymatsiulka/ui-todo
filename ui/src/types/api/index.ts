export * from './TodoResponse';
export * from './TodoRequest';
