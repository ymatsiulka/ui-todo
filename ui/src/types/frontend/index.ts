export * from './props/SpacerProps';
export * from './props/ButtonProps';
export * from './props/TodoPageStatusesProps';
