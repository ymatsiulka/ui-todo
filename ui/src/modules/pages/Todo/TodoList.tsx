import React from 'react';
import { TodoResponse } from '../../../types/api';
import Input from '../../shared/input/Input';
import { input } from 'appConstants';

interface TodoListProps {
  todos: TodoResponse[];
}

const TodoList: React.FC<TodoListProps> = ({ todos }) => (
  <ul className="todo-list-container">
    {todos.map((t: TodoResponse) => {
      return (
        <li id={t.id.toString()} data-value={t.id.toString()}>
          <Input
            name={t.id.toString()}
            label={t.name}
            type={input.checkbox}
            checked={t.isCompleted}
            onChange={(e) => {}}
          />
        </li>
      );
    })}
  </ul>
);

export default TodoList;
