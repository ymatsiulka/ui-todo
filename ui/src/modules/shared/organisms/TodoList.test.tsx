import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { TodoPageStatuses } from 'types/frontend';
import { useAppDispatch } from 'hooks';
import TodoList from './TodoList';

const mockDispatch = jest.fn();
const mockUseAppDispatch = useAppDispatch as jest.Mock;

jest.mock('hooks', () => ({
  useAppDispatch: jest.fn(),
}));

jest.mock('modules/shared/moleculas/TodoItem', () =>
  jest.fn(({ onDragStartHandler, onDragEnterHandler, onDragEndHandler }) => (
    <li
      data-testid="li"
      onDragStart={onDragStartHandler}
      onDragEnter={onDragEnterHandler}
      onDragEnd={onDragEndHandler}
    />
  )),
);

beforeEach(() => {
  mockDispatch.mockClear();
  mockUseAppDispatch.mockReturnValueOnce(mockDispatch);
});

test('TodoList should render properly', async () => {
  // Arrange
  // Act
  const { asFragment, getByTestId } = render(
    <TodoList
      todoItems={[
        {
          id: 1,
          order: 1,
          name: 'do smth',
          isCompleted: true,
        },
      ]}
      todoPageStatus={TodoPageStatuses.Completed}
    />,
  );

  const listItem = getByTestId('li');
  fireEvent.dragStart(listItem);
  fireEvent.dragEnter(listItem);
  fireEvent.dragEnd(listItem);

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
