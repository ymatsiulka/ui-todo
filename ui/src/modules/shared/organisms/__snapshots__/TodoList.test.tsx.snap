// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render properly 1`] = `
<DocumentFragment>
  <ul
    class="todoList"
    title="ui-todo-list"
  >
    <li
      data-testid="li"
    />
  </ul>
</DocumentFragment>
`;
